### PR TITLE
CSP-865 Made title properties dynamic in area of interest

### PR DESCRIPTION
**Pull request by ManishAgarwalDfE**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs [hunk 1]
```diff
@@ -14,32 +15,30 @@
...
 {
     static readonly string YourAmbassadorProfileUrl = Guid.NewGuid().ToString();
+    private static Guid memberId = Guid.NewGuid();
+    private EditAreaOfInterestController sut = null!;
+    private Mock<IOuterApiClient> _outerApiMock = null!;
+    private Mock<IValidator<SubmitAreaOfInterestModel>> _validatorMock = null!;
+    private GetMemberProfileResponse memberProfileResponse = null!;
+    private readonly List<Profile> profiles = new()
+    {
+        new Profile { Id = 1, Description = "Networking at events in person", Category = "Events", Ordering = 1 },
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs:31]: Put category titles like 'Events', 'Promotions' into a static variable, and use this variable in the checks

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/EditAreaOfInterestController.cs [hunk 2]
```diff
@@ -65,10 +65,13 @@ public async Task<EditAreaOfInterestViewModel> GetAreaOfInterests(CancellationTo
...
         var profilesResult = await _outerApiClient.GetProfilesByUserType(MemberUserType.Apprentice.ToString(), cancellationToken);
 
-        editAreaOfInterestViewModel.Events = SelectProfileViewModelMapping(profilesResult.Profiles.Where(x => x.Category == Category.Events).ToList(), memberProfiles.Profiles);
+        editAreaOfInterestViewModel.FirstSectionInterests = SelectProfileViewModelMapping(profilesResult.Profiles.Where(x => x.Category == Category.Events).ToList(), memberProfiles.Profiles);
 
-        editAreaOfInterestViewModel.Promotions = SelectProfileViewModelMapping(profilesResult.Profiles.Where(x => x.Category == Category.Promotions).ToList(), memberProfiles.Profiles);
+        editAreaOfInterestViewModel.SecondSectionInterests = SelectProfileViewModelMapping(profilesResult.Profiles.Where(x => x.Category == Category.Promotions).ToList(), memberProfiles.Profiles);
         editAreaOfInterestViewModel.YourAmbassadorProfileUrl = Url.RouteUrl(SharedRouteNames.YourAmbassadorProfile)!;
+
+        editAreaOfInterestViewModel.FirstSectionTitle = "Events";
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web/Controllers/EditAreaOfInterestController.cs:73]: Put the Section titles into a constants class and refer to that.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs [hunk 3]
```diff
@@ -183,4 +132,79 @@ public void SelectProfileViewModelMapping_ReturnsSelectProfileViewModelList(bool
...
+        var result = sut.Get(CancellationToken.None);
+        var viewResult = result as ViewResult;
+        var viewModel = viewResult!.Model as EditAreaOfInterestViewModel;
+
+        // Assert
+        using (new AssertionScope())
+        {
+            Assert.Multiple(() =>
+            {
+                Assert.That(viewModel!.FirstSectionTitle, Is.EqualTo("Events"));
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs:152]: use the constants for the section titles as suggested lower down

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs (hunk 1 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs]: Done

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/EditAreaOfInterestController.cs (hunk 2 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/EditAreaOfInterestController.cs]: Done

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs (hunk 3 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/EditAreaOfInterestControllerTests/EditAreaOfInterestControllerGetTests.cs]: Done

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**